source/app.js

```diff
@@ -0,0 +1,64 @@
+import browser from 'detect-browser'
+import translations from './translations'
+
+(function () {
+  let options = INSTALL_OPTIONS
+  const {localStorage} = window
+  const DAY_DURATION = 1000 * 60 * 60 * 24
+  const now = new Date()
+  const weekAgo = new Date(now - DAY_DURATION * 7)
+
+  const language = window.navigator.language || window.navigator.userLanguage || 'en'
+  const [messageLabel, moreLabel] = translations[language] || translations[language.substring(0, 2)] || translations.en
+
+  const appElement = document.createElement('cloudflare-app')
+  appElement.setAttribute('app-id', 'a-better-browser')
+
+  function updateElement () {
+    const browserVersion = parseFloat(browser.version.match(/(\d.)\./))
+    const browserMinimum = options[browser.name] || 0
+
+    const outdated = browserVersion < browserMinimum
+    const seenRecently = localStorage.cfBetterBrowserDismissedAt && weekAgo >= new Date(parseInt(localStorage.cfBetterBrowserDismissedAt, 10))

```

Comment:
It is interesting. Generally we're targeting evergreen browsers and IE10+ for the platform as a app installer. Our general rule is that an app cannot prevent a page from loading because it accessed a new API in a breaking way. Some apps have started with preambles that omit browser checks in favor of feature checks.

```js
(function () {
  if (!window.addEventListener) return // Check for IE9+
  
  document.addEventListener("DOMContentLoaded", updateElements)
}())
```